### PR TITLE
Wdt 247# discover custom security mbean

### PR DESCRIPTION
**Pull request by CarolynRountree**
Discover custom MBean attributes. Right now for specifically discovering Security Provider MBeans. Add custom conversion routine for discovered values.

Tested with added unit test and discovering IDCS domain in offline and online.

### CODE REVIEW

**Comment by CarolynRountree** [comment]
discovered

File: core/src/main/python/wlsdeploy/tool/discover/custom_folder_helper.py [hunk 1]
```diff
@@ -0,0 +1,516 @@
...
+        return add_value
+
+    def convert(self, value, value_type):
+        """
+        Public function to convert the value with value_type to a model compatible value
+        :param value: Value to be converted into the appropriate model data type
+        :param value_type: data type of the value
+        :return: converted data type and value
+        """
+        _method_name = '___convert_method'
```
Comment (rakillen) [inline core/src/main/python/wlsdeploy/tool/discover/custom_folder_helper.py]: = 'convert'

File: core/src/main/python/wlsdeploy/tool/util/mbean_utils.py [hunk 2]
```diff
@@ -9,11 +9,10 @@
 
 from oracle.weblogic.deploy.exception import BundleAwareException
 
-from wlsdeploy.aliases.aliases import Aliases
 from wlsdeploy.aliases.wlst_modes import WlstModes
+from wlsdeploy.exception.expection_types import ExceptionType
```
Comment (rakillen) [inline core/src/main/python/wlsdeploy/tool/util/mbean_utils.py]: Not used?

File: core/src/main/resources/oracle/weblogic/deploy/messages/wlsdeploy_rb.properties [hunk 3]
```diff
@@ -707,6 +709,43 @@ WLSDPLY-06709=Discover Multi-tenant Topology
...
 
+# custom_folder_helper.py and
+WLSDPLY-06750=Unable to determine the provider type for security provider {0} custom mbean {1}
+WLSDPLY-06751=Discover {0} Custom MBean {1}
+WLSDPLY-06752=Adding {0} to {1} Custom MBean {2}
+WLSDPLY-06753=Unable to discover {0} Custom MBean {1} with name {2}. The MBean was not loaded \
+  into the current WLST session.
+WLSDPLY-06754=Attribute {0} for custom mbean {1} is read-only
+WLSDPLY-06755=Attribute {0} value is default value {2} for custom mbean {1}
+WLSDPLY-06756={0}={1} custom mbean {2}
```
Comment (rakillen) [inline core/src/main/resources/oracle/weblogic/deploy/messages/wlsdeploy_rb.properties]: mbean -> MBean, also previous line

**Pull request merged**